src/acktest/adoption.py

```diff
@@ -0,0 +1,144 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from abc import abstractmethod, ABC
+from dataclasses import dataclass, asdict
+from typing import Dict, Tuple
+
+from .k8s import resource as k8s
+from .resources import random_suffix_name
+
+ADOPTED_RESOURCE_GROUP = "services.k8s.aws"
+ADOPTED_RESOURCE_VERSION = "v1alpha1"
+ADOPTED_RESOURCE_PLURAL = "adoptedresources"
+ADOPTED_RESOURCE_KIND = "AdoptedResource"
+
+ADOPTED_CONDITION_NAME = "ACK.Adopted"
+
+@dataclass(frozen=True)
+class AdoptedResourceAWSIdentifier:
+    """ Represents the base AWS identifier spec fields from the adopted resource CRD.
+
+    Additional keys need to be configured as allowlisted as part of the 
+    controller generator.
+    """
+    additionalKeys: Dict[str, str]
```

Comment:
here and below, please ensure `snake_case` is used for field and variable names, per Python convention. You've used `snake_case` for most things here... just a few missed ones like here and `nameOrID` below.

Reply:
Can't do this. These values are used as the keys when serializing to and from JSON for k8s specs. We need them to have the correct case to match the CRD.